### DISCUSSION TITLE
for native module in mini app

### DISCUSSION THREAD

**Discussion by HongQuang231** [callstack/repack#839] [Q&A] [answered]
can I create a native module for the mini app and when I run the host app and mini app then the native module in the mini app can work?

**Comment by jbroma** [comment, marked as answer]
only JS part of the miniapp can be federated.

If you want to include native code from the the mini-app I'd suggest creating a separate lib with the native module and including it in both host-app and mini-app as a dependency

including it in host-app enables you to use the module by the mini-app in the federated configuration.

including it in mini-app allows for development and actual use in the federated configuration.

please remember to mark that lib with native module as a shared in the MF config
